exist-core/src/main/java/org/exist/test/runner/XMLTestRunner.java

```diff
@@ -78,18 +78,53 @@ public class XMLTestRunner extends AbstractTestRunner {
 
     /**
      * @param path The path to the XML file containing the tests.
+     * @param doc the document node contaning the tests
+     * @param info the parsed test information
      * @param parallel whether the tests should be run in parallel.
-     *
-     * @throws InitializationError if the test runner could not be constructed.
      */
-    public XMLTestRunner(final Path path, final boolean parallel) throws InitializationError {
+
+    XMLTestRunner(final Path path, final Document doc, final XMLTestInfo info, final boolean parallel) {
         super(path, parallel);
+        this.doc = doc;
+        this.info = info;
+    }
+
+    /**
+     * Construct a test runner appropriate to the kind of XML file supplied. This can be
+     * 1. XML formatter XQuery tests of eXist-DB
+     * 2. XQTS test XML files from the XQTS test suite
+     *
+     * @param path The path to the XML file containing the tests.
+     * @param parallel whether the tests should be run in parallel
+     * @return the appropriate kind of test runner determined after some exploratory probing of the XML file
+     * @throws InitializationError if the file doesn't contain a format of XML tests we know about
+     */
+    public static AbstractTestRunner getRunner(final Path path, final boolean parallel) throws InitializationError {
+
+        /*
+         * TODO (AP) pull this (factory) method out into AbstractTestRunner, merge with the getRunner() in XSuite.java
+         * so we end up with a number of clean sibling subclasses of AbstractTestRunner.
+         */
+        final Document doc;
+        final XMLTestInfo info;
+        final XQTSTestRunner.XQTSTestSuite xqtsTestSuite;
```

Comment:
please check scope

Reply:
I've taken the test runner changes out, they will go into a separate PR.